src/main/java/io/dockstore/provision/ICGCStorageClientPlugin.java

```diff
@@ -72,40 +69,64 @@ public void setConfiguration(Map<String, String> map) {
         }
 
         public boolean downloadFrom(String sourcePath, Path destination) {
-            String client = "/icgc/dcc-storage/bin/dcc-storage-client";
-            if (config.containsKey(DCC_CLIENT_KEY)) {
-                client = config.get(DCC_CLIENT_KEY);
-            }
+            String clientKey = getKey();
 
             // ambiguous how to reference synapse files, rip off these kinds of headers
             String prefix = "icgc://";
             if (sourcePath.startsWith(prefix)){
                 sourcePath = sourcePath.substring(prefix.length());
+            } else {
+                System.err.println("File prefix not handled by this plugin.");
+                return false;
             }
 
-            // default layout saves to original_file_name/object_id
-            // file name is the directory and object id is actual file name
-            String downloadDir = destination.getParent().toFile().getAbsolutePath();
-            String bob = client + " --quiet" + " download" + " --object-id " + sourcePath + " --output-dir " + downloadDir + " --output-layout id";
-            Utilities.executeCommand(bob, System.out, System.err);
-
-            // downloaded file
-            String downloadPath = new File(downloadDir).getAbsolutePath() + "/" + sourcePath;
-            System.out.println("download path: " + downloadPath);
-            Path downloadedFileFileObj = Paths.get(downloadPath);
             try {
-                Files.move(downloadedFileFileObj, destination);
-                return true;
-            } catch (IOException ioe) {
-                System.err.println(ioe.getMessage());
-                throw new RuntimeException("Could not move input file: ", ioe);
+                Files.createDirectories(destination);
+            } catch (IOException e) {
+                System.err.println("Could not create destination directory: " + destination.toFile().getAbsolutePath());
+                return false;
             }
+            String command = "docker run -e ACCESSTOKEN=" + clientKey + " --mount type=bind,source=" + destination.toFile().getAbsolutePath() + ",target=/data overture/score bin/score-client --quiet download --object-id " + sourcePath + " --output-dir /data";
+            try {
+                runCommand(command);
+            } catch (Exception e) {
+                e.printStackTrace();
+                return false;
+            }
+            return true;
         }
 
         public boolean uploadTo(String destPath, Path sourceFile, Optional<String> metadata) {
             throw new UnsupportedOperationException("ICGC storage client upload not implemented yet");
         }
 
+        private void runCommand(String command) throws IOException, InterruptedException {
+            ProcessBuilder builder = new ProcessBuilder();
+            builder.command(command.split("\\s+"));
```

Comment:
I wouldn't think this would work for arguments with spaces, e.g., a directory with spaces, but you're passing an argument with spaces already and presumably that works. Hmm